src/v/bytes/bytes.h

```diff
@@ -134,3 +134,24 @@ bytes_type_eq::operator()(const bytes& lhs, const iobuf& rhs) const {
     }
     return true;
 }
+
+inline bytes operator^(bytes_view a, bytes_view b) {
+    if (unlikely(a.size() != b.size())) {
```

Comment:
this is exactly `vassert`

also why not use std::transform?

Reply:
vassert doesn't throw. how are they the same?

Reply:
i see. good point.

Reply:
ah forgot about std::transform, thx. fixed